UploadOrganizationLogo/__tests__/handler.test.ts

```diff
@@ -0,0 +1,160 @@
+
+import {
+  IAzureApiAuthorization,
+  UserGroup
+} from "io-functions-commons/dist/src/utils/middlewares/azure_api_auth";
+import { IAzureUserAttributes } from "io-functions-commons/dist/src/utils/middlewares/azure_user_attributes";
+
+import { NonNegativeInteger } from "italia-ts-commons/lib/numbers";
+import {
+  EmailString,
+  NonEmptyString,
+  OrganizationFiscalCode
+} from "italia-ts-commons/lib/strings";
+
+import { toAuthorizedCIDRs } from "io-functions-commons/dist/src/models/service";
+
+import { MaxAllowedPaymentAmount } from "io-functions-commons/dist/generated/definitions/MaxAllowedPaymentAmount";
+
+import { left, right } from "fp-ts/lib/Either";
+import * as reporters from "italia-ts-commons/lib/reporters";
+import { Logo } from "../../generated/api-admin/Logo";
+import { UploadOrganizationLogoHandler } from "../handler";
+
+const mockContext = {
+  // tslint:disable: no-console
+  log: {
+    error: console.error
+  }
+} as any;
+
+afterEach(() => {
+  jest.resetAllMocks();
+  jest.restoreAllMocks();
+});
+
+const anOrganizationFiscalCode = "01234567890" as OrganizationFiscalCode;
+const anEmail = "test@example.com" as EmailString;
+
+const aServiceId = "s123" as NonEmptyString;
+const someSubscriptionKeys = {
+  primary_key: "primary_key",
+  secondary_key: "secondary_key"
+};
+
+const aService = {
+  authorizedCIDRs: toAuthorizedCIDRs([]),
+  authorizedRecipients: new Set([]),
+  departmentName: "IT" as NonEmptyString,
+  isVisible: true,
+  maxAllowedPaymentAmount: 0 as MaxAllowedPaymentAmount,
+  organizationFiscalCode: anOrganizationFiscalCode,
+  organizationName: "AgID" as NonEmptyString,
+  requireSecureChannels: false,
+  scope: "NATIONAL",
+  serviceId: aServiceId,
+  serviceName: "Test" as NonEmptyString,
+  version: 1 as NonNegativeInteger,
+  ...someSubscriptionKeys
+};
+
+const someUserAttributes: IAzureUserAttributes = {
+  email: anEmail,
+  kind: "IAzureUserAttributes",
+  service: aService
+};
+
+const aUserAuthenticationDeveloper: IAzureApiAuthorization = {
+  groups: new Set([UserGroup.ApiServiceRead, UserGroup.ApiServiceWrite]),
+  kind: "IAzureApiAuthorization",
+  subscriptionId: aServiceId,
+  userId: "u123" as NonEmptyString
+};
+
+const aLogoPayload: Logo = {
+  logo: "base64-logo-img" as NonEmptyString
+};
+
+describe("UploadOrganizationLogo", () => {
+  it("should respond with 201 if logo upload was successfull", async () => {
+    const apiClientMock = {
+      uploadOrganizationLogo: jest.fn(() =>
+        Promise.resolve(right({ status: 201 }))
+      )
+    };
+
+    const uploadOrganizationLogoHandler = UploadOrganizationLogoHandler(
+      apiClientMock as any
+    );
+    const result = await uploadOrganizationLogoHandler(
+      mockContext,
+      aUserAuthenticationDeveloper,
+      undefined as any, // not used
+      someUserAttributes,
+      anOrganizationFiscalCode,
+      aLogoPayload
+    );
+
+    expect(apiClientMock.uploadOrganizationLogo).toHaveBeenCalledTimes(1);
```

Comment:
Do we need this test? This function is mocked, so we expect the test to just fail if it's not called